packages/core/src/workflow/sleep.ts

```diff
@@ -0,0 +1,94 @@
+import type { StringValue } from 'ms';
+import ms from 'ms';
+import { EventConsumerResult } from '../events-consumer.js';
+import { WorkflowSuspension } from '../global.js';
+import type { WorkflowOrchestratorContext } from '../private.js';
+import { withResolvers } from '../util.js';
+
+export function createSleep(ctx: WorkflowOrchestratorContext) {
+  return async function sleepImpl(param: StringValue | Date): Promise<void> {
+    const { promise, resolve } = withResolvers<void>();
```

Comment:
Would be easy to return a `SleepResult` object now that it doesn't go through the serialization layer.

Reply:
That would be an API breaking change though no? I'm still conflicted on `SleepResult` imo. It's basically just `Void`